core/bin/zksync_api/src/bin/providers/proxy_liquidity_provider.rs

```diff
@@ -0,0 +1,84 @@
+use std::{collections::HashMap, fs::read_to_string, path::Path, str::FromStr};
+
+use actix_web::{web, HttpResponse, Result};
+use tokio::sync::Mutex;
+use zksync_api::fee_ticker::CoinGeckoTypes::AssetPlatform;
+use zksync_config::ETHClientConfig;
+use zksync_types::{Address, TokenInfo};
+use zksync_utils::remove_prefix;
+
+use super::proxy_utils::{proxy_request, ProxyState};
+
+const TESTNET_PLATFORM_ID: &str = "testnet";
+const TESTNET_PLATFORM_NAME: &str = "Rootstock Testnet";
+const TESTNET_PLATFORM_SHORTNAME: &str = "testnet";
+const ROOTSTOCK_PLATFORM_ID: &str = "rootstock";
+
+async fn handle_get_asset_platforms() -> Result<HttpResponse> {
+    Ok(HttpResponse::Ok().json(vec![AssetPlatform {
+        id: String::from(TESTNET_PLATFORM_ID),
+        chain_identifier: Some(ETHClientConfig::from_env().chain_id as i64),
+        name: String::from(TESTNET_PLATFORM_NAME),
+        shortname: String::from(TESTNET_PLATFORM_SHORTNAME),
+    }]))
+}
+
+fn load_tokens(path: impl AsRef<Path>) -> Result<Vec<TokenInfo>, serde_json::Error> {
+    serde_json::from_str(&read_to_string(path).unwrap())
+}
+
+async fn handle_get_coin_contract(
+    path: web::Path<(String, String)>,
+    data: web::Data<AppState>,
+) -> HttpResponse {
+    let (_, contract_address) = path.into_inner();
+    let testnet_token_address = Address::from_str(remove_prefix(&contract_address)).unwrap();
+
+    let testnet_token = data
+        .testnet_tokens
+        .iter()
+        .find(|token| token.address.eq(&testnet_token_address));
+    let mainnet_token = match testnet_token {
+        Some(testnet_token) => data
+            .mainnet_tokens
+            .iter()
+            .find(|token| token.symbol.eq(&testnet_token.symbol)),
+        None => None,
+    };
+
+    let url = format!(
+        "https://api.coingecko.com/api/v3/coins/{}/market_chart/{}",
```

Comment:
Do we want to hard-code the URL here? Otherwise, shouldn't we add a config param? If so, do you think it's worth it?

Reply:
I don't know, we seem to be hardcoding things here anyway and:
1. this is unlikely to change
2. if it does we will probably have to change other things too

Reply:
Yes, I agree, so let's just define it in one [place only](https://github.com/rsksmart/rif-rollup/pull/101#discussion_r1336821752) 👍🏼

Reply:
done